doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,324 @@
+A high level overview of the hardware support hierarchy and terms used in the
+Zephyr documentation can be seen in :ref:`hw_support_hierarchy`.
+
+For SoC porting, the most important terms are:
+
+- SoC: the exact system on a chip the board's CPU is part of
+- SoC series: a smaller group of tightly related SoCs
+- SoC family: a wider group of SoCs with similar characteristics
+- CPU Cluster: a cluster of one or more CPU cores.
+- CPU core: a particular CPU in an architecture
+- Architecture: an instruction set architecture
+
+Architecture
+============
+
+See :ref:`architecture_porting_guide`.
+
+
+Create your SoC directory
+*************************
+
+Each SoC must have a unique name. Use the official name given by the SoC vendor
+and check that it's not already in use. In some cases someone else may have
+contributed a SoC with identical name. If the SoC name is already in use, then
+you should probably improve the existing SoC instead of creating a new one.
+The script ``list_hardware`` can be used to retrieve a list of all SoCs known
+in Zephyr, for example ``./scripts/list_hardware.py --soc-root=. --socs`` from
+the Zephyr base directory for a list of names that are already in use.
+We will use ``soc1`` in the following
+
+Start by creating the directory ``zephyr/soc/<VENDOR>/soc1``, where
+``<VENDOR>`` is your vendor subdirectory.
+
+.. note::
+  A ``<VENDOR>`` subdirectory is mandatory if contributing your SoC
+  to Zephyr, but if your SoC is placed in a local repo, then any folder
+  structure under ``<your-repo>/soc`` is permitted.
+  The ``<VENDOR>`` subdirectory must match a vendor defined in the list in
+  :zephyr_file:`dts/bindings/vendor-prefixes.txt`. If the SoC vendor does not
+  have a prefix in that list, then one must be created.
+
+.. note::
+
+  The SoC directory name does not need to match the name of the SoC.
+  Multiple SoC can even defined be in one directory. In Zephyr, SoCs are often
+  organized in sub-folders under a common SoC Family or SoC Series tree.
+
+Your SoC directory should look like this:
+
+.. code-block:: none
+
+   soc/<VENDOR>/<soc-name>
+   ├── soc.yml
+   ├── soc.h
+   ├── CMakeLists.txt
+   ├── Kconfig
+   ├── Kconfig.soc
+   └── Kconfig.defconfig
+
+Replace ``<soc-name>`` with your SoC's name.
+
+
+The mandatory files are:
+
+#. :file:`soc.yml`: a YAML file describing the high-level meta data of the
+   SoC such as:
+   - SoC name: the name of the SoC name
+   - CPU clusters: CPU clusters if the SoC contains one or more clusters
+   - SoC series: the SoC series to which the SoC belong
+   - SoC family: the SoC family to which the series belong
+
+#. :file:`soc.h`: a header file which can be used to describe or provide
+   configuration macros for the SoC. The :file:`soc.h` will often be included in
+   drivers, sub-systems, boards, and other source code found in Zephyr.
+
+#. :file:`Kconfig.soc`: the base SoC configuration which defines a Kconfig SoC
```

Comment:
> which defines a Kconfig SoC symbol

which one? CONFIG_SOC?

Reply:
No, a `config SOC_<SOC_NAME>` setting.
Will try to rephrase.